src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,505 @@
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+         
+        m_service->call_function("aggregate",
+                              args.dump(),
+                              [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
+            handle_response(error, value, completion_block);
+        });
+    } catch (const std::exception& e) {
+        return completion_block({}, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+    }
+}
+
+void RemoteMongoCollection::count(const std::string& filter_json,
+                                  uint64_t limit,
+                                  std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+{
+    try {
+        auto base_args = m_base_operation_args;
+        base_args.push_back({ "query", nlohmann::json::parse(filter_json) });
+        
+        if (limit != 0) {
+            base_args.push_back({ "limit", limit });
+        }
+        
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+
+        m_service->call_function("count",
+                                 args.dump(),
+                                 [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
+            handle_count_response(error, value, completion_block);
+        });
+    } catch (const std::exception& e) {
+        return completion_block(0, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+    }
+}
+
+void RemoteMongoCollection::count(const std::string& filter_json,
+                                  std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+{
+    count(filter_json, 0, completion_block);
+}
+
+void RemoteMongoCollection::insert_many(std::vector<std::string> documents,
+                                        std::function<void(std::map<uint64_t, std::string>, util::Optional<AppError>)> completion_block)
+{
+     try {
+
+         auto documents_array = nlohmann::json::array();
+         for (std::string& document_json : documents) {
+             documents_array.push_back(nlohmann::json::parse(document_json));
+         }
+
+         auto base_args = m_base_operation_args;
+         base_args.push_back({ "documents", documents_array });
+         auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+
+         m_service->call_function("insertMany",
+                                  args.dump(),
+                                  [completion_block](util::Optional<AppError> error, util::Optional<std::string> value) {
+             if (value && !error) {
+                 try {
+                     auto json = nlohmann::json::parse(value.value());
+                     auto inserted_ids = json.at("insertedIds").get<std::vector<nlohmann::json>>();
+                     auto mapped_oids = std::map<uint64_t, std::string>();
```

Comment:
It looks like this should just be a vector? The key of the map is just the index, so it doesn't need to be there at all.

Reply:
changed to use vector